awspub/common.py

```diff
@@ -16,3 +22,35 @@ def _split_partition(val: str) -> Tuple[str, str]:
         partition = "aws"
         resource = val
     return partition, resource
+
+
+def _get_regions(s3_bucket_region: str, regions_configured: List[str]) -> List[str]:
+    """
+    Get the regions in the config. Either configured in the image or sns
```

Comment:
I would write somethink independent from image or sns because this is now a common function. eg:
```
Get a list of region names querying the `region_to_query` for all regions and then filtering by `regions_allowlist`. 
If no `regions_allowlist` is given, all queried regions are returned for the current partition.
If `regions_allowlist` is given, all regions from that list are returned if the listed region exist in the current partition. Eg. `us-east-1` listed in `regions_allowlist` won't be returned if the current partition is `aws-cn`.
```